ivy/functional/frontends/sklearn/metrics/_classification.py

```diff
@@ -17,3 +17,28 @@ def accuracy_score(y_true, y_pred, *, normalize=True, sample_weight=None):
         ret = ret / y_true.shape[0]
         ret = ret.astype("float64")
     return ret
+
+
+@to_ivy_arrays_and_back
+def precision_score(y_true, y_pred, *, average="binary", sample_weight=None):
+    # TODO: implement sample_weight
+    y_type = type_of_target(y_true)
+    if y_type.startswith("multilabel"):
+        true_positives = ivy.count_nonzero(
+            ivy.equal(y_true, y_pred).astype("int64"), axis=0
+        )
+        all_positives = ivy.count_nonzero(y_pred, axis=0)
+    else:
+        true_positives = ivy.count_nonzero(
+            ivy.equal(y_true, y_pred).astype("int64"), axis=1
+        )
+        all_positives = ivy.count_nonzero(y_pred)
+    if average == "binary":
+        precision = true_positives / all_positives
+    elif average == "micro":
+        precision = ivy.sum(true_positives) / ivy.sum(all_positives)
+    elif average == "macro":
+        precision = ivy.mean(true_positives / all_positives)
+    else:
+        raise ValueError("Invalid value for 'average'.")
```

Comment:
For all the value errors I would suggest a look at ivy.utils.exceptions.IvyValueError.
[IvyValueError](https://unify.ai/docs/ivy/docs/utils/ivy.utils.exceptions.html#ivy.utils.exceptions.IvyValueError)
[Exception handling](https://unify.ai/docs/ivy/overview/deep_dive/exception_handling.html)